pkg/storage/gcp/saver.go

```diff
@@ -12,6 +13,10 @@ import (
 	googleapi "google.golang.org/api/googleapi"
 )
 
+// After how long we consider an "in_progress" metadata key stale,
+// due to failure to remove it.
+const inProgressStaleThreshold = 2 * time.Minute
```

Comment:
Would a user ever want to configure this threshold?

Reply:
That's a good question. I could perhaps see it mattering on extremely slow networks, where it could take more than 2 minutes to upload the zip? That seems a little tenuous, though, I suppose.

I don't think I have a strong opinion one way or another.

Reply:
For Athens that's actually not an unreasonable usecase.

Reply:
I have added a singleflight option for it and updated the PR, but the way it works is kinda ugly, stemming from the fact that the config option is a singleflight backend option that needs to be accessed in the storage backend in `Save()`.

Happy to rework/remove/separate out as necessay, given feedback.

Reply:
Also, please review the config changes closely, in case I added it incorrectly.